domains/primitives/auto-id/src/lib.rs

```diff
@@ -96,13 +102,41 @@ impl TryFrom<x509_parser::prelude::Validity> for Validity {
     }
 }
 
+#[derive(Debug, Decode, Encode, TypeInfo, PartialEq, Eq, Clone)]
+pub struct SubjectDistinguishedName {
+    /// Common name encoded as a DER vector.
+    pub common_name: DerVec,
+    /// Raw value of this subject distinguished name.
+    pub raw: DerVec,
```

Comment:
why do we need the raw bytes here?
We do not need to hold this info. We already hold the entire cert as raw so if we offchain client needs it, they can get it from the cert itself